File: fsms-backend-api/src/main/java/com/fuelstation/managmentapi/fuelstation/domain/models/FuelTank.java
package com.fuelstation.managmentapi.fuelstation.domain.models;

import com.fuelstation.managmentapi.fuelgrade.domain.FuelGrade;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.Optional;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class FuelTank {
    private Long id;
    private FuelGrade fuelGrade;
    private BigDecimal currentVolume;
    private BigDecimal maxCapacity;
    private Optional<OffsetDateTime> lastRefillDate;

    public BigDecimal getAvailableVolume() {
        return this.maxCapacity.subtract(this.currentVolume);
    }

    public static FuelTank create(FuelGrade fuelGrade) {
        return new FuelTank(
            null,
            fuelGrade,
            BigDecimal.ZERO,
            BigDecimal.valueOf(35000),
            Optional.empty()
        );
    }
}

File: rest-api/src/test/kotlin/com/fuelstation/management_api/fuel-station/domain/models/FuelTank.kt
package com.fuelstation.management_api.fuel

import com.fuelstation.management_api.common.domain.FuelGrade
import java.time.LocalDate-station.domain.models

data class FuelTank(
    val id: Number, 
    val fuelGrade: FuelGrade,
    val currentVolume: Float,
    val maxCapacity: Int,
    val lastRefilDate: LocalDate
)